addon_template/_dev_tools/addon_tasks/utils.py

```diff
@@ -0,0 +1,458 @@
+        return schema.get("default", [])
+    elif schema["type"] == "number":
+        # Return the default value if specified, otherwise 0 if required
+        return schema.get("default", 0)
+    elif schema["type"] == "integer":
+        # Return the default value if specified, otherwise 0 if required
+        return schema.get("default", 0)
+    elif schema["type"] == "null":
+        # Just return None for null types
+        return None
+    elif schema["type"] == "any":
+        # return None for any type if no default
+        return schema.get("default", None)
+    else:
+        # Extend with additional types as needed
+        raise ValueError(f"Unsupported type in path {path}: {schema['type']}")
+
+
+def _get_authenticated_session(element_path, url, username, password):
+    from seeq import sdk, spy
+    spy.login(username=username, password=password, url=url, quiet=True)
+    auth_header = {'sq-auth': spy.client.auth_token}
+    items_api = sdk.ItemsApi(spy.client)
+    element_project_name = get_element_identifier_from_path(element_path)
+    response = items_api.search_items(filters=[f'name=={element_project_name}'], types=['Project'])
+    if len(response.items) == 0:
+        raise Exception(f"Could not find a project with name {element_project_name}")
+    project_id = response.items[0].id
+    requests_session = _create_requests_session()
+    return requests_session, auth_header, project_id
+
+
+def _create_requests_session():
+    import requests
+    from requests.adapters import HTTPAdapter, Retry
+    max_request_retries = 5
+    request_retry_status_list = [502, 503, 504]
+    _http_adapter = HTTPAdapter(
+        max_retries=Retry(total=max_request_retries, backoff_factor=0.5, status_forcelist=request_retry_status_list))
+    request_session = requests.Session()
+    request_session.mount("http://", _http_adapter)
+    request_session.mount("https://", _http_adapter)
+    return request_session
+
+
+def _upload_file(server_url, request_session, auth_header, project_id, source, destination):
+    from requests.exceptions import RetryError
+
+    jupyter_path = _get_jupyter_contents_api_path(server_url, project_id, destination)
+    base_name = os.path.basename(source)
+    with open(source, 'rb') as file:
+        contents = file.read() or b''
+    body = json.dumps({'path': jupyter_path.replace('//', r'/'),
+                       'content': base64.b64encode(contents).decode('ascii'),
+                       'format': 'base64',
+                       'name': base_name,
+                       'type': 'file'})
+    response = None
+    try:
+        response = request_session.put(jupyter_path, data=body, headers=auth_header, cookies=auth_header,
+                                       verify=True, timeout=60)
+    except RetryError:
+        pass
+    if response is None or response.status_code == 500:
+        _upload_directory(server_url, auth_header, request_session, project_id, destination)
```

Comment:
The issue with `watch` was when it was trying to upload a directory. Your calling function has `auth_header` and `request_session` arguments in wrong order from what the method signature was expecting.

Reply:
thanks for fixing it